code/datums/components/extradimensional_storage.dm

```diff
@@ -56,13 +78,40 @@ TYPEINFO(/datum/component/extradimensional_storage)
 	else
 		Obj.set_loc(region.turf_at(rand(3, region.width - 2), rand(3, region.height - 2)))
 
-/datum/component/extradimensional_storage/proc/on_disposing()
-	qdel(src)
+/datum/component/extradimensional_storage/storage/UnregisterFromParent()
+	UnregisterSignal(src.parent, COMSIG_ATOM_ENTERED)
+	UnregisterSignal(src.parent, COMSIG_PARENT_PRE_DISPOSING)
+	. = ..()
 
-/datum/component/extradimensional_storage/UnregisterFromParent()
-	var/obj/storage/parent = src.parent
-	src.region.move_movables_to(parent.loc)
-	region.clean_up(/turf/space, /turf/space)
-	qdel(region)
-	UnregisterSignal(parent, COMSIG_ATOM_ENTERED)
+// subtype of the component that instead handles ladders
+/datum/component/extradimensional_storage/ladder
+
+/datum/component/extradimensional_storage/ladder/Initialize(width=9, height=9, region_init_proc=null)
+	. = ..()
+	if(!istype(parent, /obj/ladder))
+		return COMPONENT_INCOMPATIBLE
+	exit = get_turf(src.parent)
+	. = ..()
+
+	var/obj/ladder/ladder = src.parent
+	var/image/I = image(ladder.icon,ladder,"[ladder.icon_state]-extra")
+	ladder.UpdateOverlays(I,"extradim")
+
+	RegisterSignal(src.parent, COMSIG_ATTACKHAND, PROC_REF(on_entered))
+	RegisterSignal(src.parent, COMSIG_PARENT_PRE_DISPOSING, PROC_REF(on_disposing))
+
+/datum/component/extradimensional_storage/ladder/on_entered(atom/movable/thing,mob/user)
+	var/obj/ladder/ladder = src.parent
+	if (istype(ladder,/obj/ladder/embed))
+		boutput(user, "You enter the gap in the wall.")
+	else
+		boutput(user, "You climb [ladder.icon_state == "ladder" ? "down" : "up"] the ladder.")
+	user.set_loc(region.turf_at(rand(3, region.width - 2), rand(3, region.height - 2)))
+
+/datum/component/extradimensional_storage/ladder/UnregisterFromParent()
+	var/obj/ladder/ladder = src.parent
+	ladder.UpdateOverlays(null,"extradim")
+	ladder.broken = FALSE
```

Comment:
`broken = FALSE`? Why here? What does it do?

Reply:
Ah. It disables normal ladder behaviour. I feel like the var name could be better in that case.

Reply:
renamed to unclimbable, hopefully that's a good enough name